### PR TITLE
Add cache state to task summaries on real runs (2nd try)

### PR DESCRIPTION
**Pull request by mehulkar**
- Original: #4225
- Bug: #4389
- Revert: #4390 

The second commit in this PR shows the fix for the bug. I'm not sure I can write a test to cover this

### CODE REVIEW

File: cli/internal/runcache/runcache.go [hunk 1]
```diff
@@ -124,26 +125,35 @@ func (tc *TaskCache) RestoreOutputs(ctx context.Context, prefixedUI *cli.Prefixe
...
+			return cache.ItemStatus{Local: false, Remote: false}, nil
 		}
 
 		if err := tc.rc.outputWatcher.NotifyOutputsWritten(ctx, tc.hash, tc.repoRelativeGlobs); err != nil {
 			// Don't fail the whole operation just because we failed to watch the outputs
 			prefixedUI.Warn(ui.Dim(fmt.Sprintf("Failed to mark outputs as cached for %v: %v", tc.pt.TaskID, err)))
 		}
 	} else {
+		// If no outputs have changed, that means we have a local cache hit.
+		cacheStatus.Local = true
```
Comment (mehulkar) [inline cli/internal/runcache/runcache.go:156]: This was the missing piece the last time around

**Review by NicholasLYang — APPROVED** [review]
I think this makes sense, but I'm not sure if I can repro locally, should we canary and test?

**Comment by mehulkar** [comment]
> but I'm not sure if I can repro locally, should we canary and test?

This is how I test locally at both commits:
```
~/build-turbo.sh && localturbo run build --skip-infer --force && localturbo run build --skip-infer && localturbo run build --skip-infer
```
- `build-turbo` is an alias to `cd && make` in my turbo repo directory
- `localturbo` is an alias to where the turbo binary is built on my system

**Comment by NicholasLYang** [comment]
Yeah I tried something similar. It worked, but I couldn't reproduce the bug when I checked out the first commit. I'll try again and confirm that the bug occurs with my local turbo version and that the second commit fixes it.

**Comment by NicholasLYang** [comment]
Hmm I still can't get a cache miss on the first commit. If you can repro on the first commit and it seems to be fixed on the second commit then I'm down to merge.

**Comment by mehulkar** [comment]
@NicholasLYang I just had some trouble reproducing also, but I was able to get it by running 20 builds in a row with:

```
$ ~/build-turbo.sh && localturbo run build --skip-infer --force && for i in {1..20}; do localturbo run build --skip-infer --output-logs=new-only; done
```

Here's my local output from `v1.8.7`, the tip of this branch, and the tip of this branch minus one commit

https://gist.github.com/mehulkar/3c312e9df2ede8800f4992caaacf6811

**Comment by mehulkar** [comment]
going to merge this as I feel pretty confident that we've identified and fixed the bug.

**Pull request merged**